marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.tsx

```diff
@@ -304,46 +322,150 @@ const SourceToEntityMap = (props) => {
         return mapStyle;
     }
 
+    //For filter search in source table
+    let searchInput: any;
+    //For Source Table
+    const [searchSourceText,setSearchSourceText] = useState('');
+    const [searchedSourceColumn,setSearchedSourceColumn] = useState('');
+    //For Entity table
+    const [searchEntityText,setSearchEntityText] = useState('');
+    const [searchedEntityColumn,setSearchedEntityColumn] = useState('');
+
+    const handleColSearch = (selectedKeys, confirm, dataIndex) => {
+        confirm();
+        if(dataIndex === 'key'){
+            setSearchSourceText(selectedKeys[0]);
+            setSearchedSourceColumn(dataIndex); 
+        } else {
+            setSearchEntityText(selectedKeys[0]);
+            setSearchedEntityColumn(dataIndex);
+        }
+      };
+
+    const handleSearchReset = (clearFilters,dataIndex) => {
+        clearFilters();
+        if(dataIndex === 'key'){
+            setSearchSourceText('');
+            setSearchedSourceColumn(''); 
+        } else {
+            setSearchEntityText('');
+            setSearchedEntityColumn('');
+        }
+    };
+
+    const getColumnFilterProps = dataIndex => ({
+        filterDropdown: ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => (
+            <div className={styles.filterContainer}>
+                <Input
+                    ref={node => {
+                        searchInput = node;
+                    }}
+                    data-testid={`searchInput-${dataIndex}`}
+                    placeholder={`Search name`}
```

Comment:
Can we update placeholder to be {`Search ${dataIndex}`} so its unique, ie Search key for source table and Search name for entity table, instead of using data-testid. We should use data-testid sparingly if there are other possible selectors.

Reply:
@bsrikan - The placeholder text is as per the design given by UX. Also, the work '**key**' is used behind the scenes for that column and user might get confused here if we mention "Search key" as placeholder because to the user the column name is '**Name**'.

Reply:
You are right. Will let it be then.